[Flaking Test][sig-api-machinery]k8s.io/kubernetes/test/integration/apiserver/discovery.discovery

### Which jobs are flaking?

* [sig-release-master-blocking#integration-master](https://testgrid.k8s.io/sig-release-master-blocking#integration-master&exclude-non-failed-tests=)

### Which tests are flaking?

* [k8s.io/kubernetes/test/integration/apiserver/discovery.discovery](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-master/2029325907562336256)
* [Triage](https://storage.googleapis.com/k8s-triage/index.html?job=ci-kubernetes-integration-masterurltest=k8s.io/kubernetes/test/integration/apiserver/discovery.discovery)

### Since when has it been flaking?

[3/5/2026, 4:10:00 AM](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-master/2029325907562336256)
[3/5/2026, 1:07:58 AM](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-master/2029280105997537280)
[3/4/2026, 11:07:59 PM](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-master/2029249906207625216)
[3/4/2026, 12:08:00 PM](https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-integration-master/2029083809835651072)

### Testgrid link

* [Testgrid](https://testgrid.k8s.io/sig-release-master-blocking#integration-master&exclude-non-failed-tests=)

### Reason for failure (if possible)

```
	F 2026-03-05 04:10:00 +0530 IST Failed
=== RUN   TestReadinessAggregatedAPIServiceDiscovery
    testse...scovery
--- FAIL: TestReadinessAggregatedAPIServiceDiscovery (18.31s)

	F 2026-03-05 01:07:58 +0530 IST Failed
=== RUN   TestReadinessAggregatedAPIServiceDiscovery
    testse...scovery
--- FAIL: TestReadinessAggregatedAPIServiceDiscovery (18.20s)

	F 2026-03-04 23:07:59 +0530 IST Failed
=== RUN   TestReadinessAggregatedAPIServiceDiscovery
    testse...scovery
--- FAIL: TestReadinessAggregatedAPIServiceDiscovery (18.03s)

	F 2026-03-04 12:08:00 +0530 IST Failed
=== RUN   TestReadinessAggregatedAPIServiceDiscovery
    testse...scovery
--- FAIL: TestReadinessAggregatedAPIServiceDiscovery (18.14s)


```

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig api-machinery
/kind flake
cc @kubernetes/release-team-release-signal

--- a/test/integration/apiserver/discovery/framework.go
+++ b/test/integration/apiserver/discovery/framework.go
@@ -46,7 +46,7 @@ const acceptV1JSON = "application/json"
 const acceptV2JSON = "application/json;g=apidiscovery.k8s.io;v=v2;as=APIGroupDiscoveryList"
 const acceptV2JSONNoPeer = "application/json;g=apidiscovery.k8s.io;v=v2;as=APIGroupDiscoveryList;profile=nopeer"
 
-const maxTimeout = 10 * time.Second
+const maxTimeout = 30 * time.Second
 
 type testClient interface {
 	kubernetes.Interface

